### PR TITLE
chore: use workspace SDK functions

### PR DESCRIPTION
**Pull request by g-linville**
This updates the dataset tools to use the workspace provider SDK.
I updated the tests so that we write and read all of the data in a single go, rather than reading pre-existing data from a test workspace.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -12,7 +13,7 @@ import (
 type elementInput struct {
 	Name        string `json:"name"`
 	Description string `json:"description"`
-	Content     string `json:"content"`
+	Contents    string `json:"contents"`
```
Comment (g-linville) [inline main.go:16]: I updated this to make it more consistent with the SDK's data types.

File: pkg/dataset/manager.go [hunk 2]
```diff
@@ -1,51 +1,55 @@
...
 
-	return Manager{datasetDir: datasetDir}, nil
+	return Manager{gptscriptClient: g}, nil
 }
 
-func (m *Manager) ListDatasets() ([]DatasetMeta, error) {
-	files, err := filepath.Glob(filepath.Join(m.datasetDir, "*"+extension))
+func (m *Manager) ListDatasets(ctx context.Context) ([]DatasetMeta, error) {
+	files, err := m.gptscriptClient.ListFilesInWorkspace(ctx, gptscript.ListFilesInWorkspaceOptions{
+		Prefix:      datasetFilePrefix,
```
Comment (thedadams) [inline pkg/dataset/manager.go]: I would be surprised if this worked in general. The intention with the `Prefix` fields (here and others) is that they are "directories." If we need more general prefixes, then I believe we would have to add support for it.

Comment (g-linville) [reply pkg/dataset/manager.go]: At least with the directory provider, this works. I have not tested it with S3 yet.

Comment (thedadams) [reply pkg/dataset/manager.go]: Interesting. It is the directory provider that I figured would not work.

**Review by thedadams — APPROVED** [review]

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**